### PR TITLE
Wip master

### PR DESCRIPTION
**Pull request by alexanderGugel**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,12 +9,24 @@
...
-The tricky thing here will be keeping the emphasis on the store and helping
-students avoid concerns in actions etc. We want to try to make these patterns
-automatic for students as they continue to practice them.
+There are two stores: `Store` and `CounterStore`. `CounterStore` inherits from
+`Store`.
+
+The global application state that we're updating is a simple counter,
+represented by a number.
+
+Our `<App />` component renders tow buttons and the actual counter number. If
```
Comment (thomastuts) [inline README.md]: 'tow' -> 'two'

**Review by thomastuts — CHANGES_REQUESTED** [review]
@alexanderGugel This looks great! 🔥 

The only thing I'm really missing here is more explicit instructions. Students should be able to complete a lab by just using the lab's instructions, even if they skipped the reading. That's currently impossible (unless they start digging into tests), because some explicit instructions are missing (e.g. what should the methods in the store be called, what key should be used to store the counter value in the state, ...).

PS: `sinon` isn't installed by default in all labs, so don't forget to add it to `package.json` in both branches, or the tests won't run properly!

**Comment by alexanderGugel** [comment]
@thomastuts I added some more instructions in the `README` concerning the specific objectives of the lab, but I'm still unsure how specific those should be. There are also some hints in the source code itself (e.g. that `CounterStore` should inherit from `Store` etc).

The tests themselves will print an error that `addListener` etc. should be a function, so I'm wondering if those should still be explicitly outlined in the README... It seems to give away quite a lot...

Thoughts?

File: README.md [hunk 2]
```diff
@@ -9,20 +9,88 @@
...
-In this lesson we're going implement a simple `<Counter />` component. We're
-going to focus on the stores and actions while building this application.
+In this lesson we're going implement a simple counter application.
+Our focus will be on the stores and actions, how they interact with each other
+and how we can wire them up to components.
 
-There are two stores: `Store` and `CounterStore`. `CounterStore` inherits from
-`Store`.
+Our counter has two buttons used for incrementing and decrementing a number.
+This is how it's going to look like:
```
Comment (thomastuts) [inline README.md]: `This is how it's going to look` _or_ `This is what it's going to look like`

File: README.md [hunk 3]
```diff
@@ -9,20 +9,88 @@
...
+The global application state that we're updating is a simple counter,
+represented by a number. The initial count will be set to `0`, which will also
+be the initial state of the `counterStore`.
+
+### Actions
+
+There are two actions: `increment()` and `decrement()`. The action creators are
+directly wired up to the global `counterStore` singleton.
+
+`increment()` and `decrement()` respectively update the `counterStore` by
```
Comment (thomastuts) [inline README.md]: I know this is unintentional, but this feels like an incomplete sentence. Maybe we can end it with '... by doing the following:', or something similar?

File: README.md [hunk 4]
```diff
@@ -9,20 +9,88 @@
...
+1. retrieving the previous state using `store.getState()`.
+2. adding / subtracting `1` from the previous state and calling
+    `store.setState(newState)`
+
+### Stores -> Components
+
+Our `<App />` component doesn't directly update the store, but subscribes to
+its updates. The store is the single source of truth of our application, but is
+completely decoupled from individual components. In other words, there is no way
+how a component can directly update a store.
```
Comment (thomastuts) [inline README.md]: '....there is no way a component can...' (remove the _how_)

_or_

'...there is no way for a component to directly update the store'

**Review by thomastuts — CHANGES_REQUESTED** [review]
@alexanderGugel This looks great, instructions are super clear now. Some really minor nits, but after that, we'll be good to merge!

**Comment by alexanderGugel** [comment]
@thomastuts Fixed in 30ff804

**Comment by thomastuts** [comment]
@alexanderGugel Awesome, thanks! 🎉

**Pull request merged**